VMEncryption/main/handle.py

```diff
@@ -1080,20 +1080,25 @@ def enable_encryption():
 
                     bek_util.store_bek_passphrase(encryption_config, extension_parameter.passphrase)
 
-                #If PrivatePreview.ConfidentialEncryptionTempDisk is set to True in public settings
-                #then temp disk encryption will happen for CVM type
-                  
-                public_settings = get_public_settings()
-                confidential_encryption_tempdisk = public_settings.get("PrivatePreview.ConfidentialEncryptionTempDisk")
+                #Temp disk encryption will happen for CVM type               
                 encryptResourceDisk = False
                 if security_Type==CommonVariables.ConfidentialVM:
-                    if confidential_encryption_tempdisk:
-                        encryptResourceDisk = True
-                    else: 
-                        message = "Resource disk not encrypted. It is confidential VMs, but PrivatePreview.ConfidentialEncryptionTempDisk is not set"
-                        logger.log(msg=message)
+                    public_settings = get_public_settings()
+                    no_confidential_encryption_tempdisk = public_settings.get("NoConfidentialEncryptionTempDisk")
+                    no_confidential_encryption_tempdisk_flag = False
+                    msg = ""
+                    if no_confidential_encryption_tempdisk.__class__.__name__ in ['str','bool']:
+                        if no_confidential_encryption_tempdisk.__class__.__name__ == 'str' and no_confidential_encryption_tempdisk.lower() == "true":
+                            no_confidential_encryption_tempdisk_flag=True
```

Comment:
You could add a log message that the NoConfidentialEncryptionTempDisk is set to true by the customer. It'd come handy in handling customer reported incidents.

Reply:
done